prow/cmd/deck/main.go

```diff
@@ -1296,8 +1296,10 @@ func handleRerun(prowJobClient prowv1.ProwJobInterface, createProwJob bool, cfg
 				}
 				login, err := goa.GetLogin(r, ghc)
 				if err != nil {
-					http.Error(w, fmt.Sprintf("Error retrieving GitHub login: %v", err), http.StatusInternalServerError)
 					l.WithError(err).Errorf("Error retrieving GitHub login")
+					if _, err = w.Write([]byte("needs_login")); err != nil {
```

Comment:
I like the idea of using a sentinel value. One minor improvement would be to leverage a specific http status code that explicitly *means* the user is not logged in; the status code returned in these cases is typically a [401 Unauthorized](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401).  Then on the client-side, you can simply check this http code instead of string.

```suggestion
					http.Error(w, fmt.Sprintf("Error retrieving GitHub login: %v", err), http.StatusUnauthorized)
```